packages/react/src/components/RoomPreview/RoomPreview.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useRef } from "react";
+
+type RoomPreviewProps = {
+  /** Image to show while the room preview is loading. */
+  loadingUrl?: string;
+  /** URL of the room preview. */
+  previewUrl?: string;
+  /** Custom styles. */
+  style?: React.CSSProperties;
+};
+
+const RoomPreview: React.FC<RoomPreviewProps> = ({
+  previewUrl,
+  loadingUrl,
+  style,
+}) => {
+  const container = useRef<HTMLDivElement>(null);
+  const currImg = useRef<HTMLImageElement>(getDefaultImage(loadingUrl));
+
+  // Initialization
+  useEffect(() => {
+    if (container.current) {
+      container.current?.appendChild(currImg.current);
+      if (loadingUrl) {
+        currImg.current.src = loadingUrl;
+        currImg.current.addEventListener("error", () => {
+          if (currImg.current.src !== loadingUrl) {
+            currImg.current.src = loadingUrl;
+          }
+        });
+      }
+    }
+  }, [container.current]);
```

Comment:
Also, you don't need to use `container.current` as dependency as that has no effect.